### DISCUSSION TITLE
A try to use the forked published `ast-grep-wasm` package.

### DISCUSSION THREAD

**Discussion by LoTwT** [ast-grep/ast-grep#1053] [Show and tell]
## description

I copy the codes of `ast-grep-wasm` and publish the `ag-wasm`. It's also built by the `wasm-pack`. I use some scripts to redesign the dist structure and update some `exports` fields in the `package.json` for better type support.

Then I make a demo playground referring to [ast-grep.github.io](https://github.com/ast-grep/ast-grep.github.io) and [sxzz/ast-explorer](https://github.com/sxzz/ast-explorer) that can view `tree-sittier` ast using `ag-wasm`. The demo URL is <https://ast.ayingott.me/>.

## detail

- `wasm-pack build --target web`
- use `vite-plugin-wasm`
- config optimizeDeps like

```ts
// nuxt/config.ts
{
  ...,
  vite: {
    plugins: [wasm()],

    optimizeDeps: {
      include: ["ag-wasm > web-tree-sitter-wasm-bindgen"],
      exclude: ["ag-wasm"],
    },
  },
}
```

Then use the `ag-wasm` just like the `ast-grep-wasm` in the `ast-grep.github.io`.

## else

Hi, @HerringtonDarkholme . Could you give it some advice? Hope it could be a `throw a brick to attract a jade` for publishing the `ast-grep-wasm` in the future.

**Comment by HerringtonDarkholme** [comment]
Hi @LoTwT, thanks for the trial! Would you like to share the repo here so more people can take it as a reference? Thanks!

**Comment by LoTwT** [comment]
😳 Sorry. I forget to add related links.

The links of repositories that I create:

- [ag-wasm](https://github.com/lotwt/ag-wasm)
- [ast-explorer](https://github.com/lotwt/ast-explorer)
